osu.Server.Spectator/Startup.cs

```diff
@@ -80,8 +86,21 @@ protected virtual void ConfigureAuthentication(IServiceCollection services)
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, IHostApplicationLifetime lifetime)
         {
+            var shutdownManager = app.ApplicationServices.GetService<GracefulShutdownManager>();
+
+            Debug.Assert(shutdownManager != null);
+
+            // This is done here, but it happens too late in shutdown.
+            // The more important registration is in EntityStore.
+            lifetime.ApplicationStopping.Register(shutdownManager.WaitForSafeShutdown);
```

Comment:
Also don't see why this is required if https://github.com/ppy/osu-server-spectator/pull/133/files/681bed1d646394408b7f83cf9b9923dc8bf0bd95#r883196847 were applied.